strncmp warnings in  bgp_community_alias.c

<!--

*** ATTENTION ***

YOU MUST READ THIS TO HAVE YOUR ISSUE ADDRESSED

PLEASE READ AND FILL OUT THIS TEMPLATE

NEGLECTING TO PROVIDE INFORMATION REQUESTED HERE WILL RESULT IN
SIGNIFICANT DELAYS ADDRESSING YOUR ISSUE

ALWAYS PROVIDE:
- FRR VERSION
- OPERATING SYSTEM VERSION
- KERNEL VERSION

FAILURE TO PROVIDE THIS MAY RESULT IN YOUR ISSUE BEING IGNORED

FOLLOW THESE GUIDELINES:

- When reporting a crash, provide a backtrace
- When pasting configs, logs, shell output, backtraces, and other large chunks
  of text, surround this text with triple backtics

  ```
  like this
  ```

- Include the FRR version; if you built from Git, please provide the commit
  hash
- Write your issue in English

-->

---------------

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Put "x" in "[ ]" if you already tried following:
-->

[x] Did you check if this is a duplicate issue?
[x] Did you test it on the latest FRRouting/frr master branch?

Building frr 8.1 (same code is in master) with `-Wstringop-overread` yields these warnings in `bgpd/bgp_community_alias.c`:
```
gcc -std=gnu11 -DHAVE_CONFIG_H -DSYSCONFDIR=\"/etc/frr/\" -DCONFDATE=0 -I.  -I./lib/assert -I. -I./include -I./lib -I.    -Wdate-time -D_FORTIFY_SOURCE=2 -fno-omit-frame-pointer -funwind-tables -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wpointer-arith -Wbad-function-cast -Wwrite-strings -Wundef -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers        -g -O2 -ffile-prefix-map=/home/ubuntu/git/packages/frr/frr=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -c -o bgpd/bgp_conditional_adv.o bgpd/bgp_conditional_adv.c
bgpd/bgp_community_alias.c: In function ‘bgp_ca_alias_hash_cmp’:
bgpd/bgp_community_alias.c:60:17: warning: ‘strncmp’ specified bound 8228 exceeds source size 8192 [-Wstringop-overread]
   60 |         return (strncmp(ca1->alias, ca2->alias, sizeof(struct community_alias))
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bgpd/bgp_community_alias.c: In function ‘bgp_ca_community_hash_cmp’:
bgpd/bgp_community_alias.c:43:17: warning: ‘strncmp’ specified bound 8228 exceeds source size 36 [-Wstringop-overread]
   43 |         return (strncmp(ca1->community, ca2->community,
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   44 |                         sizeof(struct community_alias))
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

The size parameter should be the size of the specific struct member that is being compared, `alias` and `community`, and not the whole struct.

**To Reproduce**
<!--
Describe the steps to reproduce the behavior.
Be as descriptive as possible.

For example:

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
-->
Build frr.

**Expected behavior**
<!--
Write here a clear and concise description of what you expected to happen when
using the reproduction steps you provided above
-->
No warnings for `strncmp` usage.

**Screenshots**
<!--
If applicable, add screenshots to help explain your problem.
-->

**Versions**
<!--
Include your operating system type and version here

FAILURE TO PROVIDE THIS MAY RESULT IN YOUR ISSUE BEING IGNORED
-->
<!-- e.g. Fedora 24, Debian 10] -->
 - OS Version: Ubuntu Jammy 22.04 (devel)
<!-- [e.g. Linux 5.4, OpenBSD 6.6] -->
 - Kernel: Linux 5.13.0
<!-- e.g. 6.0, 7.4 -->
 - FRR Version: 8.1 (and master)

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/bgpd/bgp_community_alias.c
+++ b/bgpd/bgp_community_alias.c
@@ -40,9 +40,7 @@ static bool bgp_ca_community_hash_cmp(const void *p1, const void *p2)
 	const struct community_alias *ca1 = p1;
 	const struct community_alias *ca2 = p2;
 
-	return (strncmp(ca1->community, ca2->community,
-			sizeof(struct community_alias))
-		== 0);
+	return (strcmp(ca1->community, ca2->community) == 0);
 }
 
 static unsigned int bgp_ca_alias_hash_key(const void *p)
@@ -57,8 +55,7 @@ static bool bgp_ca_alias_hash_cmp(const void *p1, const void *p2)
 	const struct community_alias *ca1 = p1;
 	const struct community_alias *ca2 = p2;
 
-	return (strncmp(ca1->alias, ca2->alias, sizeof(struct community_alias))
-		== 0);
+	return (strcmp(ca1->alias, ca2->alias) == 0);
 }
 
 static void *bgp_community_alias_alloc(void *p)

